### ISSUE TITLE
Device doesn't cover all traits from Cpu/Cuda

### ISSUE THREAD

**Issue by jackerschott** [coreylowman/dfdx#866]
It is hard to work with a generic type, e.g. `D: Device<f32>`, since some of the traits from `Cpu`/`Cuda` are not implemented for `Device` and sometimes not public. E.g. the `ToDtypeKernel` trait, such that I have to use something like AutoDevice when using `to_dtype` with other types than f32 and f64.
Would it be sensible to include all shared traits of `Cpu`/`Cuda` in `Device` or implement some new trait that includes everything?

**Comment by LeoDog896** [comment]
This is also partially why I proposed https://github.com/coreylowman/dfdx/issues/859, but this seems like a much better full solution to the matter.

**Comment by coreylowman** [comment]
Yeah we can add more instances of ToDtypeKernel constraints to Device.
